sdk/inc/azure/core/az_mqtt5.h

```diff
@@ -208,6 +210,11 @@ typedef struct
    * @brief The publish request ID.
    */
   int32_t id;
+
+  /**
+   * @brief Publish acknowledgement (PUBACK) reason code. Indicates success or reason for failure.
+   */
+  int32_t puback_reason;
```

Comment:
Humm do we need puback_ inside puback struct?

Reply:
Removed the `puback_` suffix and replaced with `reason_code`